tests/unit/test_build_env.py

```diff
@@ -41,19 +42,21 @@ def run_with_build_env(script, setup_script_contents,
                 link_collector=link_collector,
                 selection_prefs=selection_prefs,
             )
-            build_env = BuildEnvironment()
 
-            try:
+            with global_tempdir_manager():
+                build_env = BuildEnvironment()
             ''' % str(script.scratch_path)) +
         indent(dedent(setup_script_contents), '    ') +
-        dedent(
-            '''
+        indent(
+            dedent(
+                '''
                 if len(sys.argv) > 1:
                     with build_env:
                         subprocess.check_call((sys.executable, sys.argv[1]))
-            finally:
-                build_env.cleanup()
-            ''')
+                '''
+            ),
+            '    '
+        )
```

Comment:
Would this be more readable?

```python
indent(
    dedent(
        setup_script_contents +
        '''
        blah blah blah
        '''
    ),
    '    ',
)
```

Reply:
Note that this needs setup_script_contents to have the same indentation as the string here.

Reply:
Marking as resolved since we don't want to require the indentation of these two separate items to match.